File: javascript和css学习/bubble_sort.js
function bubbleSort(a){
    var n = a.length;
    var flag = n-1;
    while(flag){
        var i = flag;
        flag = 0;
        for(var j=0;j<i;j++){
            if(a[j] > a[j+1]){
                var temp = a[j];
                a[j] = a[j+1];
                a[j+1] = temp;
                flag = j;
            }
        }
    }
}

File: bubble_sort.py
def bubble_sort(a):
    n = len(a)
    flag = n-1
    while flag:
        i = flag
        flag = 0
        for j in range(i):
            if a[j] > a[j+1]:
                a[j],a[j+1] = a[j+1],a[j]
                flag = j